packages/loader/src/MaterialLoader.ts

```diff
@@ -32,73 +33,91 @@ class MaterialLoader extends Loader<Material> {
-                  new Vector3((<IVector3>value).x, (<IVector3>value).y, (<IVector3>value).z)
-                );
-                break;
-              case "Vector4":
-                materialShaderData.setVector4(
-                  key,
-                  new Vector4((<IVector4>value).x, (<IVector4>value).y, (<IVector4>value).z, (<IVector4>value).w)
-                );
-                break;
-              case "Color":
-                materialShaderData.setColor(
-                  key,
-                  new Color((<IColor>value).r, (<IColor>value).g, (<IColor>value).b, (<IColor>value).a)
-                );
-                break;
-              case "Float":
-                materialShaderData.setFloat(key, <number>value);
-                break;
-              case "Texture":
-                texturePromises.push(
-                  // @ts-ignore
-                  resourceManager.getResourceByRef<Texture2D>(<IAssetRef>value).then((texture) => {
-                    materialShaderData.setTexture(key, texture);
-                  })
-                );
-                break;
-              case "Boolean":
-                materialShaderData.setInt(key, value ? 1 : 0);
-                break;
-              case "Integer":
-                materialShaderData.setInt(key, Number(value));
-                break;
-            }
+          if (shaderRef) {
+            resolve(
+              resourceManager
+                // @ts-ignore
+                .getResourceByRef<Shader>(<IAssetRef>shaderRef)
+                .then((shaderObject) => this.getMaterialByShader(materialSchema, shaderObject, engine))
+            );
+          } else {
+            // compatible with 1.2-pre version material schema
+            const shaderObject = Shader.find(shader);
+            resolve(this.getMaterialByShader(materialSchema, shaderObject, engine));
           }
+        })
+        .catch(reject);
+    });
+  }
 
-          for (let i = 0, length = macros.length; i < length; i++) {
-            const { name, value } = macros[i];
-            if (value == undefined) {
-              materialShaderData.enableMacro(name);
-            } else {
-              materialShaderData.enableMacro(name, value);
-            }
-          }
+  private getMaterialByShader(materialSchema: IMaterialSchema, shader: Shader, engine: Engine): Promise<Material> {
+    const { name, shaderData, macros, renderState } = materialSchema;
 
-          parseProperty(material, "renderState", renderState);
+    const material = new Material(engine, shader);
+    material.name = name;
 
-          return Promise.all(texturePromises).then(() => {
-            resolve(material);
-          });
-        })
-        .catch(reject);
+    const texturePromises = new Array<Promise<Texture2D>>();
+    const materialShaderData = material.shaderData;
+    for (let key in shaderData) {
+      const { type, value } = shaderData[key];
+
+      switch (type) {
+        case "Vector2":
```

Comment:
push these variables into  `EEngineLoadType`, and consume them uniformly in the editor, and delete `transformValues.ts` in editor

Reply:
done